node/tests/GlideClusterClient.test.ts

```diff
@@ -1966,4 +1966,316 @@ describe("GlideClusterClient", () => {
             }
         },
     );
+    describe("GlideClusterClient - AZAffinity Read Strategy Test", () => {
+        it.each([ProtocolVersion.RESP2, ProtocolVersion.RESP3])(
+            "should route GET commands to all replicas with the same AZ using protocol %p",
+            async (protocol) => {
+                const az = "us-east-1a";
+                const n_replicas = 6;
+                const GET_CALLS = 3 * n_replicas;
+                const get_cmdstat = `calls=${GET_CALLS / n_replicas}`;
+
+                let client_for_config_set;
+                let client_for_testing_az;
+
+                try {
+                    // Stage 1: Configure nodes
+                    client_for_config_set =
+                        await GlideClusterClient.createClient(
+                            getClientConfigurationOption(
+                                cluster.getAddresses(),
+                                protocol,
+                            ),
+                        );
+
+                    // Skip test if version is below 8.0.0
+                    if (cluster.checkIfServerVersionLessThan("8.0.0")) {
+                        console.log(
+                            "Skipping test: requires Valkey 8.0.0 or higher",
+                        );
+                        return;
+                    }
+
+                    await client_for_config_set.customCommand([
+                        "CONFIG",
+                        "RESETSTAT",
+                    ]);
+                    await client_for_config_set.customCommand(
+                        ["CONFIG", "SET", "availability-zone", az],
+                        { route: "allNodes" },
+                    );
+
+                    // Stage 2: Create AZ affinity client and verify configuration
```

Comment:
```typescript
const readFrom: ReadFrom = "AZAffinity";

client_for_testing_az =
    await GlideClusterClient.createClient(
        getClientConfigurationOption(
            cluster.getAddresses(),
            protocol,
            {
                readFrom,
                clientAz: az,
            },
        ),
    );
```
Currentlly it erroring because it cant assign the type

Reply:
it can't assign the readFrom type? couldn't see that error when I ran the tests
and your suggestion includes setting it up outside the client creation? that would resolve it?

Reply:
Its not an error, it allows it, but its not right typescript, it shows red, we just not enforce it.
or outside or adding 
```typescript
await GlideClusterClient.createClient(
    getClientConfigurationOption(
        cluster.getAddresses(),
        protocol,
        {
            readFrom: "AZAffinity" as ReadFrom,
            clientAz: az,
        },
    ),
);